meta/defined.rb

```diff
@@ -7,3 +7,13 @@
   mutation 'defined?(nil)'
   mutation 'true'
 end
+
+Mutant::Meta::Example.add :defined? do
+  source 'defined?(@foo)'
+
+  singleton_mutations
+  mutation 'defined?(nil)'
+  mutation 'defined?(foo)'
+  mutation 'true'
```

Comment:
I would do this as a separate PR, but should the `defined?` mutator also emit `false`? It's interesting that it is only true. Is there a reason for that?

Reply:
I think this was a deliberate decision as `singleton_mutations` never emits `false`:

```
irb(main):001:0> defined?(@a)
=> nil
irb(main):002:0> @a = "foo"
=> "foo"
irb(main):003:0> defined?(@a)
=> "instance-variable"
```

emitting `true` is also wrong in that light ;) I'll add a commit to improve this.

Reply:
Ah, I guess since we are emitting `nil` we already have something falsey.

I actually think `true` was a good mutation--don't we normally replace predicates with `true` and `false`? (or maybe there is an issue for this somewhere.

Or are you thinking it's just too hard to cover normal `defined?` mutations because it's hard to have both states?

Reply:
oh! nevermind. I see that `defined?` never returns `true` normally. Ok, cool, good call, ignore me. I misunderstood what you were saying for a moment. Nice improvement.

Reply:
no worries, `defined?` is weird in other ways.

It actually gives information about the AST, but when the AST references a constant, than it also checks the presence of that constant. Ruby....